File: trunxifeizhuan_client/Assets/AssetsLibrary/AssetManager/ResourceUrl.lua
local EXTERNAL_PATH = "external/";
local SCENE_PATH = "scene/";
local FX_PATH = "fx/";
local ROLE_PATH = "role/";
local UI_ROLE_PATH = "ui_role/";
local TEXTURE_PATH = "texture/";
local SOUND_PATH = "sound/";
local Application = UnityEngine.Application;
local RuntimePlatform = UnityEngine.RuntimePlatform;
local mFile = System.IO.File;

local ResourceUrl = 
{
	buff_icon = TEXTURE_PATH .. "buff_icon/",
	camp_dungeon_icon = TEXTURE_PATH .. "camp_dungeon_icon/",
	face = TEXTURE_PATH .. "face/",
	goods_icon = TEXTURE_PATH .. "goods_icon/",
	role_icon = TEXTURE_PATH .. "role_icon/",
	skill_icon = TEXTURE_PATH .. "skill_icon/",
	seed_icon = TEXTURE_PATH .. "seed_icon/",
};

local FILE_FLAG = "file:///";
local mGameStreamingAssetsPath = nil;
local mGameStreamingAssetsRawPath = nil;
local mGameStreamingAssetsExternalPath = nil;

local mGamePersistentDataPath = nil;
local mGamePersistentDataRawPath = nil;
local mGamePersistentDataExternalPath = nil;

local mPersistentDataPath = Application.persistentDataPath;
local mPersistentDataRawPath = mPersistentDataPath .. "/StreamingAssets/";
local mPersistentDataExternalPath = mPersistentDataPath .. "/StreamingAssets/external/";

if Application.platform == RuntimePlatform.Android then
	mGameStreamingAssetsPath = Application.streamingAssetsPath;
	mGamePersistentDataPath = FILE_FLAG .. mPersistentDataPath;
else
	mGameStreamingAssetsPath = FILE_FLAG .. Application.streamingAssetsPath;
	mGamePersistentDataPath = FILE_FLAG .. mPersistentDataPath;
end

mGameStreamingAssetsRawPath = mGameStreamingAssetsPath.."/";
mGameStreamingAssetsExternalPath = mGameStreamingAssetsPath.."/external/";

mGamePersistentDataRawPath = mGamePersistentDataPath .. "/StreamingAssets/";
mGamePersistentDataExternalPath = mGamePersistentDataPath .. "/StreamingAssets/external/";

function ResourceUrl.GetExternalUrl(path)
	if mFile.Exists( mPersistentDataExternalPath .. path ) then
		return mGamePersistentDataExternalPath .. path;
	end
 	return mGameStreamingAssetsExternalPath .. path;
end

function ResourceUrl.GetUrl(path)
	if mFile.Exists( mPersistentDataRawPath .. path ) then
		return mGamePersistentDataRawPath .. path;
	end
 	return mGameStreamingAssetsRawPath .. path;
end


function ResourceUrl.GetExternalPath(assetName)
	return EXTERNAL_PATH..assetName;
end

function ResourceUrl.GetRolePath()
	return ROLE_PATH;
end

function ResourceUrl.GetUIRolePath()
	return UI_ROLE_PATH;
end

function ResourceUrl.GetScenePath()
	return SCENE_PATH;
end

function ResourceUrl.GetFXPath(assetName)
	return FX_PATH..assetName;
end

function ResourceUrl.GetSoundPath()
	return SOUND_PATH;
end

return ResourceUrl;

File: trunwanzhe_art/Assets/Plugins/Assets/Scripts/Com/Game/Utils/ResourceURL.cs
using Assets.Plugins.Assets.Scripts.Com.Game.Utils;
using System.Collections.Generic;
using System.IO;
using UnityEngine;

namespace Assets.Scripts.Com.Game.Utils
{
    public static class ResourceURL
    {
        public static string RAW_STREAMING_ASSETS { get; private set; }

        public static string PERSISTENT_STREAMING_ASSETS { get; private set; }

        public const string FILE_FLAG = "file:///";

        public static bool sUsePersistentPath = false;

        static ResourceURL()
        {
            if (Application.platform == RuntimePlatform.Android)
            {
                RAW_STREAMING_ASSETS = Application.streamingAssetsPath;
                PERSISTENT_STREAMING_ASSETS = FILE_FLAG + Application.persistentDataPath;
            }
            else
            {
                RAW_STREAMING_ASSETS = FILE_FLAG + Application.streamingAssetsPath;
                PERSISTENT_STREAMING_ASSETS = FILE_FLAG + Application.persistentDataPath;
            }
        }

        public static string GetRawPath(string path)
        {
            return RAW_STREAMING_ASSETS + "/" + path;
        }

        public static string GetPersistentPath(string path)
        {
            return PERSISTENT_STREAMING_ASSETS + "/StreamingAssets/" + path;
        }

        public static string GetPersistentFilePath(string path)
        {
            return Application.persistentDataPath + "/StreamingAssets/" + path;
        }

        public static bool CheckPersistentFileExists(string path)
        {
            return File.Exists(GetPersistentFilePath(path));
        }

        public static string GetAssetPath(string path)
        {
            if (sUsePersistentPath)
            {
                return GetPersistentPath(path);
            }

            return GetRawPath(path);
        }

    }
}
